definitions/npm/lodash_v4.x.x/flow_v0.63.x-/lodash_v4.x.x.js

```diff
@@ -2945,8 +2945,8 @@ declare module "lodash/fp" {
     omitBy<A, T: { [id: any]: A }>(predicate: OPredicate<A>, object: T): Object;
     pick(props: Array<string>): (object: Object) => Object;
     pick(props: Array<string>, object: Object): Object;
-    pickAll(props: Array<string>): (object: Object) => Object;
-    pickAll(props: Array<string>, object: Object): Object;
+    pickAll(props: string | $ReadOnlyArray<string>): (object: Object) => Object;
+    pickAll(props: string | $ReadOnlyArray<string>, object: Object): Object;
```

Comment:
Can you show documentation of where a `string` is accepted? Looking at the source code and docs, it seems like only arrays are allowed.

Reply:
Hey @AndrewSouthpaw 

Here you go! [Lodash Pick](https://lodash.com/docs/4.17.11#pick)

And I just committed new changes... it is `pick` rather than `pickAll`!! :smile:

Reply:
You modified the `lodash/fp` definitions, the base `lodash` definitions lies near line 1244.

https://github.com/flow-typed/flow-typed/blob/a643a3e570dbb07e0cb97973254fa8f7cd23ebc6/definitions/npm/lodash_v4.x.x/flow_v0.63.x-/lodash_v4.x.x.js#L1244-L1245

So I believe the changes we want here are:

```js
pick(...props: $ReadOnlyArray<string>, object: Object): Object;
pick(...props: $ReadOnlyArray<string>): (object: Object) => Object;
pick(props: $ReadOnlyArray<string>, object: Object): Object;
pick(props: $ReadOnlyArray<string>): (object: Object) => Object;
```

```
_.pick('a', { a: 1, b, 2 })
_.pick('a')
_.pick(['a'], { a: 1, b, 2 })
_.pick(['a'])
```

Could you add a few unit tests as well?

Reply:
@pascalduez We actually need **both** `fp` and classic!

Added classic `pick` and Unit Tests

Looking good now?? :smile:

Reply:
@edoardo-bluframe Still not, check the sample I gave you above. 

Spread arguments typing like `...props: $ReadOnlyArray<string>` allows for any number of string arguments, which `pick` support. `_.pick('a', 'b', 'c')`. It's not the same thing as `props: string | $ReadOnlyArray<string>`. 
So declaring the different signatures as above is preferred.

Reply:
@pascalduez We had one spread already

I added the other

Better now?? 😄

Reply:
@edoardo-bluframe You kept the `string |` which must be removed :)

Reply:
@pascalduez It **must** be good now yeah?? :smile: